buildpack.md

```diff
@@ -658,31 +658,31 @@ If the environment variable file name ends in `.prepend`, then the value of the
 - Earlier buildpacks' environment variable file contents MUST precede later buildpacks' environment variable file contents.
 - Environment variable file contents originating from the same buildpack MUST be sorted alphabetically ascending by associated layer name.
-- Environment variable file contents originating in the same layer MUST be sorted such that file contents in `<layers>/<layer>/env/` precede file contents in `<layers>/<layer>/env.build/` or `<layers>/<layer>/env.launch/`.
+- **Environment variable file contents originating in the same layer MUST be sorted such that file contents in `<layers>/<layer>/env/` precede file contents in `<layers>/<layer>/env.build/` or `<layers>/<layer>/env.launch/` which must precede file contents in `<layers>/<layer>/env.launch/<process>/`.**
 
 ##### Override
 
 If the environment variable file name ends in `.override`, then the value of the environment variable MUST be the file contents.
 For that environment variable value,
 - Later buildpacks' environment variable file contents MUST override earlier buildpacks' environment variable file contents.
 - For environment variable file contents originating from the same buildpack, file contents that are later (when sorted alphabetically ascending by associated layer name) MUST override file contents that are earlier.
-- Environment variable file contents originating in the same layer MUST be sorted such that file contents in `<layers>/<layer>/env.build/` or `<layers>/<layer>/env.launch/` override file contents in `<layers>/<layer>/env/`.
+- **Environment variable file contents originating in the same layer MUST be sorted such that file contents in `<layers>/<layer>/env.launch/<process>/` override file contents in `<layers>/<layer>/env.build/` or `<layers>/<layer>/env.launch/` which override file contents in `<layers>/<layer>/env/`.**
 
 ##### Default
 
 If the environment variable file name ends in `.default`, then the value of the environment variable MUST only be the file contents if the environment variable is empty.
 For that environment variable value,
 - Earlier buildpacks' environment default variable file contents MUST override later buildpacks' environment variable file contents.
 - For default environment variable file contents originating from the same buildpack, file contents that are earlier (when sorted alphabetically ascending by associated layer name) MUST override file contents that are later.
-- Default environment variable file contents originating in the same layer MUST be sorted such that file contents in `<layers>/<layer>/env/` override file contents in  `<layers>/<layer>/env.build/` or `<layers>/<layer>/env.launch/`.
+- **Default environment variable file contents originating in the same layer MUST be sorted such that file contents in `<layers>/<layer>/env/` override file contents in  `<layers>/<layer>/env.build/` or `<layers>/<layer>/env.launch/` which override file contents in `<layers>/<layer>/env.launch/<process>/`.**
```

Comment:
I think this one is backwards (both here and in the current spec). Wouldn't we want defaults from the more specific case to override the more general case?

Reply:
I agree. The whole idea is to set something only if nothing is already set. It seems to me that a single layer attempting to set the default would be expecting the most specific `default` to win.

Reply:
paging at @sclevine

Reply:
I originally thought this was inaccurate. After talking with @sclevine I realize it is accurate. If we wanted to change this behavior for default env vars I think that would be separate conversation and doesn't need to happen as part of this PR.

Reply:
@ekcasey is there a tl;dr from your discussion with Stephen?

Reply:
TLDR: This is how it works today. Setting multiple defaults is a bit of a weird edge case anyways. It would be slightly complicated to change b/c of the current implementation (the current behavior is easy/natural given the order in which we process env dirs). But we certainly could change it if we were motivated.

I feel strongly that any discussion of changing this behavior should be decoupled from the process-specific env vars discussion.